tutorials/nservicebus-101/lesson-2/tutorial.md

```diff
@@ -0,0 +1,182 @@
+ 1. Add the NServiceBus NuGet package to the Messages project.
+ 1. In the **ClientUI** project, add a reference to the **Messages** project.
+
+
+### Create a message
+
+In order to better organize messages, we'll create our first command in a folder called **Commands**.
+
+ 1. In the **Messages** project, create a new folder called **Commands**.
+ 1. In the **Commands** folder, add a new class named `PlaceOrder`.
+ 1. Mark `PlaceOrder` as `public` and implement `ICommand`.
+ 1. Add a public property of type `string` named `OrderId`.
+
+NOTE: The .NET Framework contains another interface named `ICommand` in the `System.Windows.Input` namespace. Be sure that if you use tooling to resolve the namespace, you select `NServiceBus.ICommand`. Most of the types you will need will reside in the `NServiceBus` namespace.
+
+When complete, your `PlaceOrder` class should look like the following:
+
+snippet:PlaceOrder
+
+
+### Create a handler
+
+Now that we've defined a message, we can create a corresponding message handler. Rather than create a new endpoint, let's start by just handling the message locally within the **ClientUI** endpoint.
+
+ 1. In the **ClientUI** project, create a new class named `PlaceOrderHandler`.
+ 1. Mark the handler class as public, and implement the `IHandleMessages<PlaceOrder>` interface.
+ 1. Add a logger instance, which will allow you to take advantage of the same logging system used by NServiceBus. This has an important advantage over `Console.WriteLine()`: the messages you write with the logger will be written to the log file in addition to the console. Use this code to add the logger instance to your handler class:
+    ```cs
+    static ILog logger = LogManager.GetLogger<PlaceOrderHandler>();
+    ```
+ 1. Within the `Handle` method, use the logger to record the receipt of the `PlaceOrder` message, including the value of the `OrderId` message property:
+    ```cs
+    logger.Info($"Received PlaceOrder, OrderId = {message.OrderId}");
+    ```
+ 1. Since everything we have done in this handler method is synchronous, return `Task.CompletedTask`.
+
+When complete, your `PlaceOrderHandler` class should look like this:
+
+snippet:PlaceOrderHandler
+
+
+### Send a message locally
+
+Now we have a message and a handler to process it. Let's send that message!
+
+In the **ClientUI** project, we are currently stopping the endpoint when we press the Enter key. Instead, let's create a run loop that will allow us to be a little more interactive, so that we can use the keyboard to decide whether to send a message or quit.
+
+Add the following method to the **Program.cs** file:
+
+snippet:RunLoop
+ 
+Let's take a closer look at the case when we want to place an order. Since the `PlaceOrder` command is just a class, we can instantiate it normally, supplying a unique value for the `OrderId`. Then, after logging the details, we can send it with the `SendLocal` method.
```

Comment:
> Since the `PlaceOrder` command is just a class, we can instantiate it normally, supplying a unique value for the `OrderId`.

In order to create the `PlaceOrder` command simply create an instance of the `PlaceOrder` class and supply a unique value for the `OrderId`.